### ISSUE TITLE
Modularising and refactoring server.R 

### ISSUE THREAD

**Issue by fontikar** [traitecoevo/infinitylists#86]
**Which parts of server.R can be potentially modularised**
In other words, code removed from server.R, put in its own .R or in global.R (where appropriate)

- Returning empty table if no records are found in target
- Update selectize input for genus and family 
- Creating a single statement of what has been found in target area
- In target vs in buffer zone
- Reactive of creating circle polygon
- Lat Long filter in ala_data

 **Which parts of server.R can be refactored?** 
 In other words, improved upon so server.R contains less processing code
 
 - [ ] Processing of `filtered_data()` - can occur after galah_download
     - [ ] Sorting N in descending order 
     - [ ] Creating and sorting datatable from result in target area
          - [x] #87

- [ ] Saving of CSV
    - [x] Renaming of columns  - can occur after galah_download
    - [x] Creation of URL https://biocache.ala.org.au/occurrences/
    - [x] Renaming of columns

**Comment by fontikar** [comment]
Take URL creation out of downloadhandler and display table formatting and put in download script.
